src/components/code/index.tsx

```diff
@@ -14,41 +15,202 @@ import styles, {
+            maxHeightOfCode={
+              popupProps.height || popupProps.maxHeight
+                ? `calc(${popupProps.height ?? popupProps.maxHeight} - 143px)`
+                : '100%'
+            }
+            {...props}
+          />
+        )}
+        BriefComponent={() => (
+          <CodeSnippet
+            content={content}
+            contentToShow={contentLines
+              .slice(0, CONTENT_UPPER_BOUND)
+              .join('\n')}
+            title={title}
+            language={language}
+            copyButton={copyButton}
+            downloadButton={downloadButton}
+            downloadCallback={downloadCallback}
+            wrapLongLines={wrapLongLines}
+            {...props}
+          />
+        )}
+      />
+    );
+
+  // if it's not expandable then render it Normaly
+  return (
+    <CodeSnippet
+      content={content}
+      title={title}
+      language={language}
+      copyButton={copyButton}
+      downloadButton={downloadButton}
+      downloadCallback={downloadCallback}
+      wrapLongLines={wrapLongLines}
+      {...props}
+    />
+  );
+};
+
+export default Code;
+
+// Normal Code Component
+interface CodeSnippetProps
+  extends Omit<CodeProps, 'popupProps' | 'expandable'> {
+  contentToShow: string;
+  maxHeightOfCode?: string;
+}
+const CodeSnippet: FC<CodeSnippetProps> = ({
+  title,
+  content,
+  contentToShow,
+  language,
+  copyButton,
+  downloadButton,
+  downloadCallback,
+  wrapLongLines,
+  showLineNumbers,
+  copyCallback,
+  maxHeightOfCode,
+  ...props
+}) => {
   const theme = useTheme<ITheme>();
 
+  return (
+    <Flex width="100%" sx={{ ...styles }} height="100%">
+      <Flex width="100%" sx={{ ...codeHeaderStyles }}>
+        <Box flexGrow={1} ml="8px" my={1}>
+          {title}
+        </Box>
+        {downloadButton && (
+          <DownloadButton
+            content={content}
+            downloadCallback={downloadCallback}
+            title={title}
+          />
+        )}
+        {copyButton && (
+          <CopyButton content={content} copyCallback={copyCallback} />
+        )}
+      </Flex>
+      <Flex
+        width="100%"
+        variant="code"
+        height={maxHeightOfCode}
+        {...props}
+        p={0}
+      >
+        <SyntaxHighlighter
+          wrapLongLines={wrapLongLines}
+          showLineNumbers={showLineNumbers}
+          lineNumberStyle={{
+            ...lineNumberStyles,
+            background: theme.colors.grayShade1,
+          }}
+          language={language}
+          customStyle={{
+            ...boxStyles,
+            paddingLeft: showLineNumbers ? '0px' : '20px',
+          }}
+        >
+          {contentToShow}
+        </SyntaxHighlighter>
+      </Flex>
+    </Flex>
+  );
+};
+
+// Download Button
+const DownloadButton: FC<
+  Pick<CodeProps, 'content' | 'downloadCallback' | 'title'>
+> = ({ content, downloadCallback, title }) => {
+  const download = () => {
+    if (downloadCallback) {
+      downloadCallback();
```

Comment:
I know that this is a piece of old code, but since you're in the context: don't we need to call the `downloadCallback` in all cases? Now it's either save to file or call callback. 
Shouldn't it just save even if the callback is there?
@giardiv

Reply:
I see in the code somewhere we are using just the `donwloadButton` prop without specifying `downloadCallback`.
https://github.com/logicalclocks/hopsworks-front/blob/1025acf20ec578558d8ca509694651d6bb5bc158/src/pages/project/training-dataset/forms/Features/FeatureTransformations/ViewTransformationPopup.tsx#L34-L45